mm2src/mm2_main/tests/mm2_tests/mm2_tests_inner.rs

```diff
@@ -7927,3 +7950,342 @@ fn test_eth_swap_negotiation_fails_maker_no_fallback() {
+/// run cargo test with '--ignored' option
+/// to use trezor emulator add '--features trezor-udp' option to cargo test params
+#[test]
+#[ignore]
+#[cfg(all(not(target_arch = "wasm32")))]
+fn test_withdraw_from_trezor_segwit_no_rpc() {
+    use common::wait_until_ms;
+
+    let ticker = "tBTC-segwit";
+    let coin_conf = json!({
+        "coin": ticker,
+        "name": "tbitcoin",
+        "fname": "tBitcoin",
+        "rpcport": 18332,
+        "pubtype": 111,
+        "p2shtype": 196,
+        "wiftype": 239,
+        "segwit": true,
+        "bech32_hrp": "tb",
+        "address_format": {
+            "format": "segwit"
+        },
+        "orderbook_ticker": "tBTC",
+        "txfee": 0,
+        "estimate_fee_mode": "ECONOMICAL",
+        "mm2": 1,
+        "is_testnet": true,
+        "required_confirmations": 0,
+        "protocol": {
+            "type": "UTXO"
+        },
+        // added to params from coins
+        "derivation_path": "m/84'/1'",
+        "trezor_coin": "Testnet",
+    });
+    let mm_conf = json!({ "coins": [coin_conf] });
+
+    let ctx = mm_ctx_with_trezor(mm_conf);
+    let priv_key_policy = PrivKeyBuildPolicy::Trezor;
+    let servers: Vec<_> = [
+        "electrum1.cipig.net:10068",
+        "electrum2.cipig.net:10068",
+        "electrum3.cipig.net:10068",
+    ]
+    .iter()
+    .map(|server| json!({ "url": server }))
+    .collect();
+
+    let enable_req = json!({
+        "method": "electrum",
+        "coin": ticker,
+        "servers": servers,
+        "priv_key_policy": "Trezor",
+    });
+    let activation_params = UtxoActivationParams::from_legacy_req(&enable_req).unwrap();
+    let request: InitStandaloneCoinReq<UtxoActivationParams> = json::from_value(json!({
+        "ticker": ticker,
+        "activation_params": activation_params
+    }))
+    .unwrap();
+
+    block_on(async {
+        let init_result = init_standalone_coin::<UtxoStandardCoin>(ctx.clone(), request)
+            .await
+            .unwrap();
+        let timeout = wait_until_ms(150000);
+        loop {
+            if now_ms() > timeout {
+                panic!("{} init_standalone_coin timed out", ticker);
+            }
+
+            let status_req = InitStandaloneCoinStatusRequest {
+                task_id: init_result.task_id,
+                forget_if_finished: true,
+            };
+            let status_res = init_standalone_coin_status::<UtxoStandardCoin>(ctx.clone(), status_req).await;
+            if let Ok(status) = status_res {
+                match status {
+                    RpcTaskStatus::Ok(tx_details) => break tx_details,
+                    RpcTaskStatus::Error(e) => panic!("{} init_standalone_coin error {:?}", ticker, e),
+                    _ => Timer::sleep(1.).await,
+                }
+            } else {
+                panic!("{} could not get init_standalone_coin status", ticker)
+            }
```

Comment:
Why not use `enable_utxo_v2_electrum` helper? I think this whole test code can be reduced a lot by using helper functions and constants. Please also check other tests in this PR and try to optimize their code. This will make tests more maintainable where we change things in one place to apply for all tests.

Reply:
> I can remove it at all if it is confusing

No need to remove it if you find it useful. Thanks for the explanation I missed that you don't use mm2 instance.